plugins/vendor/webex/batch/processors/KWebexAPIDropFolderEngine.php

```diff
@@ -34,11 +34,18 @@ class KWebexAPIDropFolderEngine extends KDropFolderFileTransferEngine
 	public function watchFolder(KalturaDropFolder $dropFolder)
 	{
 		$this->initDropFolderEngine($dropFolder);
-		$recordingsList = $this->retrieveRecordingsList();
-		if ($recordingsList)
+		$paginationLink = null;
+		do
 		{
-			$this->handleRecordingsList($recordingsList);
+			$recordingsList = $this->retrieveRecordingsList($paginationLink);
+			$paginationLink = $this->webexClient->getPaginationLinkFromLastRequest();

```

Comment:
makes more sense to flip the order of the lines and first handle the requests then take the next link